### ISSUE TITLE
[explicit-module-boundary-types] disable rule for Function is exported

### ISSUE THREAD

**Issue by demoshang** [typescript-eslint/typescript-eslint#2368]
**Repro**

```JSON
{
  "rules": {
     "@typescript-eslint/explicit-function-return-type": "off",
    "@typescript-eslint/explicit-module-boundary-types": ["error"]
  }
}
```

```TS
export function test() {
  return;
}
```

**Expected Result**

I just want to disable for export function check, but enable for others

**Actual Result**

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `X.Y.Z` |
| `@typescript-eslint/parser`        | `X.Y.Z` |
| `TypeScript`                       | `X.Y.Z` |
| `ESLint`                           | `X.Y.Z` |
| `node`                             | `X.Y.Z` |
| `npm`                              | `X.Y.Z` |

**Comment by bradzacher** [comment]
Hi there!

This rule is intended to check all parts of your module's boundary.
This includes functions.

We will not be accepting an option do disable certain parts of this rule.
